makefile.shade

```diff
@@ -802,12 +812,148 @@ functions
         }
 
+        static string DefaultDropsShare(string value)
+        {
+            return value ?? (Environment.OSVersion.Platform == PlatformID.Unix ? "/aspnetci-drops" : @"\\aspnetci\drops");
+        }
+
+        string CreateBuildWithFlowId(string repo)
+        {
+            string output;
+            if (IsLinux)
+            {
+                output = Path.Combine(repo, "build.sh");
+            }
+            else
+            {
+                output = Path.Combine(repo, "build-with-flow-id.cmd");
+                File.WriteAllText(
+                    output,
+                    string.Format("set KOREBUILD_FLOWID=KOREBUILD_{0}& call build.cmd %*", repo));
+            }
+
+            return Path.GetFullPath(output);
+        }
+
         static IList<IGrouping<int, string>> GetBuildGraph()
         {
-            var repositoryLookup = new List<RepositoryInfo>();
-            foreach (var repo in GetRepositoriesToBuild())
+            var repositoryLookup = GetRepositoriesToBuild()
+                    .Select(RepoInfoFactory.Create)
+                    .ToList();
+
+            foreach (var info in repositoryLookup)
+            {
+                foreach (var item in info.DependencyNames)
+                {
+                    var dependency = repositoryLookup.Find(r => r.RepositoryNames.Contains(item));
+                    if (dependency != null)
+                    {
+                        info.Dependencies.Add(dependency);
+                    }
+                }
+            }
+
+            return repositoryLookup.GroupBy(r => r.Order, r => r.Name).OrderBy(r => r.Key).ToArray();
+        }
+
+        private class RepoInfoFactory
+        {
+            public static RepositoryInfo Create(string repo)
             {
                 var info = new RepositoryInfo { Name = repo };
+                ResolveProjectJsonProjects(info, repo);
+                ResolveMsBuildProjects(info, repo);
+                info.DependencyNames.ExceptWith(info.RepositoryNames);
+                return info;
+            }
+
+            private static void ResolveMsBuildProjects(RepositoryInfo info, string repo)
+            {
+                foreach (var sln in Directory.EnumerateFiles(repo, "*.sln"))
+                {
+                    var lines = File.ReadAllLines(sln);
+                    foreach (var line in lines)
+                    {
+                        var match = Regex.Match(line, @"^# Visual Studio (\d+)\s?");
+                        if (match == null || match.Groups.Count < 2)
+                        {
+                            continue;
+                        }
+                        int version;
+                        if (int.TryParse(match.Groups[1].Value, out version) && version >= 15)
+                        {
+                            ResolveMsBuildProject(info, sln);
+                            return;
```

Comment:
This stops after the first project it finds? Or maybe I'm not understanding this?